1.Ao importar o Data frame, o primeiro erro a se notar era o de encoding, ficando claro que o encoding padrão do pandas (utf-8) não era o necessário para importar o data frame corretamente. Testando outros encodings se fez possível chegar a conclusão que em realidade o 'latin_1' era o mais adequado. 

1.1.Após resolver o primeiro problema, apareceu outro: as informações estavam amontoadas, separadas por(;). por conta disso, o comando sep() se fez bastante útil para a visualização correta. O processo se repetiu para a importação dos dois conjuntos de dados.


In [2]:
import pandas as pd
import numpy as np
d2019 = pd.read_csv('https://raw.githubusercontent.com/Carol-ACDR/datatran2019-2020/main/datatran2019.csv',sep =';',encoding='latin_1')
d2019

id data_inversa     dia_semana   horario  uf     br     km  \
0      182210.0   2019-01-01    terça-feira  01:30:00  SP  116.0    218   
1      182211.0   2019-01-01    terça-feira  01:30:00  PR  373.0  177,3   
2      182212.0   2019-01-01    terça-feira  00:45:00  SC  101.0     16   
3      182214.0   2019-01-01    terça-feira  00:00:00  CE   20.0  429,5   
4      182215.0   2019-01-01    terça-feira  01:00:00  MG  381.0    327   
...         ...          ...            ...       ...  ..    ...    ...   
67441  266255.0   2019-10-07  segunda-feira  09:10:00  ES  101.0  270,5   
67442  266406.0   2019-11-25  segunda-feira  07:20:00  PR  116.0    125   
67443  266434.0   2019-06-03  segunda-feira  19:00:00  CE  222.0    1,4   
67444  266573.0   2019-07-13         sábado  19:35:00  PR  373.0    425   
67445  266627.0   2019-10-22    terça-feira  20:20:00  BA  324.0    608   

                municipio                       causa_acidente  \
0               GUARULHOS          Falta de Atenção à Condução   
1            PONTA GROSSA          Falta de Atenção à Condução   
2                  GARUVA                     Animais na Pista   
3               FORTALEZA  Ingestão de Substâncias Psicoativas   
4                NOVA ERA          Falta de Atenção à Condução   
...                   ...                                  ...   
67441               SERRA          Falta de Atenção à Condução   
67442  FAZENDA RIO GRANDE          Falta de Atenção à Condução   
67443             CAUCAIA   Não guardar distância de segurança   
67444              CANDOI          Defeito Mecânico no Veículo   
67445        SIMOES FILHO                       Defeito na Via   

                      tipo_acidente  ... feridos_graves ilesos ignorados  \
0       Colisão com objeto estático  ...              0      0         0   
1                  Colisão traseira  ...              0      0         0   
2       Colisão com objeto estático  ...              0      0         0   
3       Colisão com objeto estático  ...              1      0         0   
4               Colisão transversal  ...              1      1         0   
...                             ...  ...            ...    ...       ...   
67441                    Tombamento  ...              1      1         0   
67442               Colisão lateral  ...              0      1         1   
67443              Colisão traseira  ...              1      1         0   
67444     Saída de leito carroçável  ...              1      0         0   
67445  Queda de ocupante de veículo  ...              0      0         0   

      feridos veiculos      latitude     longitude  regional  delegacia  \
0           4        1  -23,46052014  -46,48772478     SR-SP     DEL6/1   
1           1        1  -25,05533957  -50,22776753     SR-PR     DEL7/3   
2           1        1      -26,1216      -48,8826     SR-SC     DEL8/3   
3           1        1      -3,76999    -38,670063     SR-CE    DEL16/1   
4           1        2      -19,7609      -43,0306     SR-MG     DEL4/3   
...       ...      ...           ...           ...       ...        ...   
67441       1        2     -20,23107     -40,27387     SR-ES    DEL12/2   
67442       1        2     -25,59505     -49,31631     SR-PR     DEL7/1   
67443       1        2     -3,736507     -38,65337     SR-CE    DEL16/1   
67444       1        1      -25,7118     -52,18374     SR-PR     DEL7/3   
67445       1        1     -12,78156     -38,41656     SR-BA    DEL10/1   

            uop  
0      UOP01/SP  
1      UOP01/PR  
2      UOP01/SC  
3      UOP02/CE  
4      UOP01/MG  
...         ...  
67441  UOP01/ES  
67442  UOP03/PR  
67443  UOP01/CE  
67444  UOP02/PR  
67445  UOP01/BA  

[67446 rows x 30 columns]

In [3]:
d2020 = pd.read_csv('https://raw.githubusercontent.com/Carol-ACDR/datatran2019-2020/main/datatran2020.csv',sep =';',encoding='latin_1')
d2020

id data_inversa    dia_semana   horario  uf     br     km  \
0      260031.0   2020-01-01  quarta-feira  01:00:00  TO  153.0  678,1   
1      260036.0   2020-01-01  quarta-feira  01:00:00  RJ  116.0    178   
2      260037.0   2020-01-01  quarta-feira  01:52:00  SC  101.0  206,9   
3      260038.0   2020-01-01  quarta-feira  01:15:00  RO  364.0    236   
4      260039.0   2020-01-01  quarta-feira  01:30:00  PR  153.0    360   
...         ...          ...           ...       ...  ..    ...    ...   
63442  334713.0   2020-12-23  quarta-feira  13:58:00  PA  316.0     62   
63443  334777.0   2020-12-19        sábado  23:00:00  PA  316.0    106   
63444  334780.0   2020-12-08   terça-feira  06:10:00  PR  116.0   90,9   
63445  334788.0   2020-12-31  quinta-feira  22:30:00  SC  101.0  146,9   
63446  334959.0   2020-11-18  quarta-feira  18:00:00  AL  104.0   90,3   

                 municipio                                     causa_acidente  \
0                   GURUPI                                   Animais na Pista   
1              NOVA IGUACU                       Falta de Atenção do Pedestre   
2                 SAO JOSE                                 Ingestão de Álcool   
3                   CACOAL                            Velocidade Incompatível   
4                 REBOUCAS                                 Ingestão de Álcool   
...                    ...                                                ...   
63442            CASTANHAL                                 Ingestão de Álcool   
63443  SANTA MARIA DO PARA  Desobediência às normas de trânsito pelo condutor   
63444            PIRAQUARA                                  Condutor Dormindo   
63445              ITAPEMA            Avarias e/ou desgaste excessivo no pneu   
63446            RIO LARGO                       Falta de Atenção do Pedestre   

                   tipo_acidente  ... feridos_graves ilesos ignorados feridos  \
0        Atropelamento de Animal  ...              5      0         0       5   
1      Atropelamento de Pedestre  ...              0      1         0       0   
2                Colisão lateral  ...              0      1         0       1   
3      Saída de leito carroçável  ...              1      0         0       1   
4                    Capotamento  ...              4      0         0       6   
...                          ...  ...            ...    ...       ...     ...   
63442           Colisão traseira  ...              0      2         0       1   
63443            Colisão frontal  ...              0      1         0       1   
63444           Colisão traseira  ...              0      1         0       2   
63445         Colisão com objeto  ...              0      0         0       0   
63446  Atropelamento de Pedestre  ...              2      0         0       2   

      veiculos      latitude     longitude  regional  delegacia       uop  
0            1  -11,77460203  -49,10744996     SR-TO   UOP01/TO       NaN  
1            1  -22,75223028   -43,4379103     SR-RJ     DEL5/1  UOP02/RJ  
2            2  -27,59193546  -48,61824557     SR-SC     DEL8/1  UOP01/SC  
3            1  -11,44624577  -61,43761218     SR-RO    DEL21/2  UOP02/RO  
4            1  -25,67503796  -50,75089805     SR-PR     DEL7/3  UOP04/PR  
...        ...           ...           ...       ...        ...       ...  
63442        2   -1,29786885  -47,93764104     SR-PA      UOP01     DEL01  
63443        2   -1,34101996  -47,55122901     SR-PA      UOP02     DEL01  
63444        2   -25,4611174  -49,08880123     SR-PR      UOP01     DEL01  
63445        1  -27,09129094  -48,61669064     SR-SC      UOP03     DEL04  
63446        1   -9,50059087  -35,81127569     SR-AL      UOP01     DEL01  

[63447 rows x 30 columns]

2.

a) Em cada um dos Data frames, o número de acidentes é representado por cada entrada (row) no data frame, então, para saber o total de acidentes em cada ano, bastava saber o tamanho do index de cada um, para isso usei o comando len().

In [547]:
Total_de_acidentes_em_2019 = len(d2019.index) 
Total_de_acidentes_em_2019

67446

In [548]:
Total_de_acidentes_em_2020 = len(d2020.index)
Total_de_acidentes_em_2020

63447

b) As variáveis são dadas a partir das colunas de cada dataframe, então para saber seu total, bastava saber quantas colunas havia, para isso usei novamente o comando len(). O resultado foi o mesmo para as duas = 30 variáveis.

In [549]:
Total_de_variaveis_em_2019 = len(d2019.columns) 
Total_de_variaveis_em_2019

30

In [550]:
Total_de_variaveis_em_2020 = len(d2020.columns) 
Total_de_variaveis_em_2020

30

c) Em 2019 as 5 cidades onde ocorreram mais acidentes nas rodovias federais foram Curitiba,Brasília,São José,Guarulhos e Serra,resultado obtido ao adicionar o argumento .head() após filtrar a ocorrência dos valores na coluna município com .value_counts(). 

Já em 2020 foram Brasilia, Curitiba, Guarulhos, São José e Palhoça. A filtragem usada foi a mesma do Data frame anterior.

Ao concatenar as duas Séries e somá-las, temos que nos dois anos as 5 cidades onde ocorreram mais acidentes nas rodovias federais foram: Brasília,Curitiba,São José,Guarulhos e Palhoça.

In [553]:
MC= d2019['municipio'].value_counts().sort_values(ascending = False).head()
MC

CURITIBA     1094
BRASILIA     1090
SAO JOSE      819
GUARULHOS     717
SERRA         633
Name: municipio, dtype: int64

In [552]:
MC2 = d2020['municipio'].value_counts().sort_values(ascending = False).head()
MC2

BRASILIA     1040
CURITIBA      945
GUARULHOS     639
SAO JOSE      629
PALHOCA       593
Name: municipio, dtype: int64

In [4]:
V1 =d2019
V2 =d2020
V3 = [V1,V2]
V4 = pd.DataFrame(pd.concat(V3))
V4['municipio'].value_counts().sort_values(ascending = False).head()

BRASILIA     2130
CURITIBA     2039
SAO JOSE     1448
GUARULHOS    1356
PALHOCA      1199
Name: municipio, dtype: int64

d) Filtrando o dataframe com os dados de 2019 para que somente fossem mostradas as linhas onde os acidentes ocorreram na Paraíba e com feridos graves acima de 0. Tinha-se assim uma resposta dada pelo tamanho da Séries sobre os acidentes graves na Paraíba, para retirar esse valor, foi utilizada a função len(), que retorna justamente o tamanho da Séries.

In [5]:
PB = len(d2019[(d2019.uf == 'PB') & (d2019.feridos_graves > 0)])
PB

414

3.
a.passo1) Para formar um Data frame com ranking de estados utilizando os dados dos dois Data frames anteriores, foi realizado primeiramente uma concatenação. Seguida da criação de duas variáveis "ACidentes Por Estado"(ACPE3,ACPE4). A primeira contendo o número de acidentes registrados de cada estado e a segunda contendo o ranking baseado nesses números. 

In [6]:
V1 =d2019
V2 =d2020
ACPE1 = [V1,V2]
ACPE2 = pd.DataFrame(pd.concat(ACPE1))
ACPE3 = pd.DataFrame(ACPE2['uf'].value_counts())
ACPE4 = ACPE3.rank(ascending = False,method = 'min')


a_passo_2) Para uma melhor visualização e funcionamento do Data frame, renomeei as colunas que antes estavam com nome de 'uf' por conta da filtragem, 'rank' pertencendo a posição e 'acidentes' pertencendo ao número total de acidentes em cada estado. Como a posição do index foi tomada pelos Estados, então a nomeei de uf.

In [7]:
ACPE4.rename(columns={'uf':'rank'},inplace = True)
ACPE5 = [ACPE4,ACPE3]
ACPE6 = pd.DataFrame(pd.concat(ACPE5,axis = 1))
ACPE6.rename(columns={'uf':'acidentes'},inplace = True)
ACPE6.index.rename('uf',inplace = True)


a_passo_3) Por fim, como proposto, ficou a transformação do Data frame para um arquivo csv.

In [13]:
ACPE6.to_csv('C:\Users\DELL\Desktop\ACPE.csv', sep =';' )
Ler = pd.read_csv(ACPE.csv',sep =';')
Ler

uf  rank  acidentes
0   MG   1.0      17083
1   SC   2.0      15645
2   PR   3.0      14876
3   RJ   4.0       8828
4   RS   5.0       8771
5   SP   6.0       8417
6   BA   7.0       6914
7   GO   8.0       6580
8   ES   9.0       5372
9   PE  10.0       5293
10  MT  11.0       4326
11  CE  12.0       3263
12  RO  13.0       3018
13  MS  14.0       2974
14  PB  15.0       2965
15  RN  16.0       2770
16  PI  17.0       2546
17  MA  18.0       2226
18  DF  19.0       2130
19  PA  20.0       1776
20  AL  21.0       1257
21  SE  22.0       1186
22  TO  23.0       1115
23  AC  24.0        554
24  RR  25.0        487
25  AP  26.0        283
26  AM  27.0        238

b) Processo extremamente semelhante ao anterior, com o diferencial de que os dados em análise agora são os dias da semana.


In [1]:
V1 =d2019
V2 =d2020
ACDS2 = pd.DataFrame(pd.concat([V1,V2]))
ACDS3 = pd.DataFrame(ACDS2['dia_semana'].value_counts())
ACDS4 = ACDS3.rank(ascending = False,method = 'min')
ACDS4.rename(columns={'dia_semana':'rank'},inplace = True)
ACDS5 = [ACDS4,ACDS3]
ACDS6 = pd.DataFrame(pd.concat(ACDS5,axis = 1))
ACDS6.rename(columns={'dia_semana':'acidentes'},inplace = True)
ACDS6.index.rename('dia_semana',inplace = True)
ACDS6.to_csv(r'ACDS.csv', sep =',' )
Ler = pd.read_csv(r'C:\Users\DELL\Desktop\ACDS.csv')
Ler

NameError: name 'd2019' is not defined

4.

a) Concatenando tanto os valores de 'causa_acidente' pertencentes aos Data frames d2019 e d2020, como também a junção de seus valores, tem-se um novo Data frame que por si só já demonstra a frequência das causas. usando head(1) podemos obter a causa mais comum de acidentes, que nesse caso, vale para as Séries.

In [560]:
V1 = d2019
V2 = d2020
C19 = d2019['causa_acidente']
C20 = d2020['causa_acidente']
V3 = pd.concat([C19,C20])
Junção = V3.value_counts()
VFinal = pd.concat([Junção,C19.value_counts(),C20.value_counts()], axis = 1, keys =['Junção','2019','2020'])
VFinal.fillna(0, inplace = True)
VFinal

Junção     2019   2020
Falta de Atenção à Condução                          47406  24992.0  22414
Desobediência às normas de trânsito pelo condutor    16899   8098.0   8801
Velocidade Incompatível                              11906   6028.0   5878
Ingestão de Álcool                                   10489   5419.0   5070
Não guardar distância de segurança                    7821   4214.0   3607
Defeito Mecânico no Veículo                           7475   3732.0   3743
Condutor Dormindo                                     4592   2483.0   2109
Pista Escorregadia                                    4201   2186.0   2015
Falta de Atenção do Pedestre                          3613   1971.0   1642
Animais na Pista                                      2932   1527.0   1405
Ultrapassagem Indevida                                2254   1173.0   1081
Defeito na Via                                        1933    973.0    960
Avarias e/ou desgaste excessivo no pneu               1700    799.0    901
Mal Súbito                                            1299    715.0    584
Restrição de Visibilidade                              976    493.0    483
Objeto estático sobre o leito carroçável               881    437.0    444
Desobediência às normas de trânsito pelo pedestre      780    392.0    388
Ingestão de álcool e/ou substâncias psicoativas...     750    394.0    356
Agressão Externa                                       726    384.0    342
Carga excessiva e/ou mal acondicionada                 681    305.0    376
Sinalização da via insuficiente ou inadequada          488    247.0    241
Fenômenos da Natureza                                  452    230.0    222
Deficiência ou não Acionamento do Sistema de Il...     347    160.0    187
Ingestão de Substâncias Psicoativas                    180     94.0     86
Reação tardia ou ineficiente do condutor                14      0.0     14
Manobra de mudança de faixa                             13      0.0     13
Transitar na contramão                                  11      0.0     11
Condutor deixou de manter distância do veículo ...      10      0.0     10
Ausência de reação do condutor                          10      0.0     10
Acessar a via sem observar a presença dos outro...       7      0.0      7
Ingestão de álcool pelo condutor                         7      0.0      7
Demais falhas mecânicas ou elétricas                     5      0.0      5
Conversão proibida                                       4      0.0      4
Desrespeitar a preferência no cruzamento                 4      0.0      4
Condutor desrespeitou a iluminação vermelha do ...       3      0.0      3
Chuva                                                    2      0.0      2
Mal súbito do condutor                                   2      0.0      2
Pedestre andava na pista                                 2      0.0      2
Transitar no acostamento                                 2      0.0      2
Demais falhas na via                                     2      0.0      2
Entrada inopinada do pedestre                            2      0.0      2
Falta de acostamento                                     2      0.0      2
Acostamento em desnível                                  1      0.0      1
Ausência de sinalização                                  1      0.0      1
Afundamento ou ondulação no pavimento                    1      0.0      1
Frear bruscamente                                        1      0.0      1
Ingestão de álcool ou de substâncias psicoativa...       1      0.0      1
Acumulo de água sobre o pavimento                        1      0.0      1
Trafegar com motocicleta (ou similar) entre as ...       1      0.0      1
Condutor usando celular                                  1      0.0      1
Acumulo de óleo sobre o pavimento                        1      0.0      1
Pedestre cruzava a pista fora da faixa                   1      0.0      1

In [561]:
print(C19.value_counts().head(1))
print(C20.value_counts().head(1))
print(Junção.head(1))
VFinal.head(1)

Falta de Atenção à Condução    24992
Name: causa_acidente, dtype: int64
Falta de Atenção à Condução    22414
Name: causa_acidente, dtype: int64
Falta de Atenção à Condução    47406
Name: causa_acidente, dtype: int64


Junção     2019   2020
Falta de Atenção à Condução   47406  24992.0  22414

a2). Já a causa mais rara, no entanto, difere bastante entre 2020 e 2019. Houve um número maior de casos raros em 2020, com ocorrência de apenas 1 deste tipo, enquanto em 2019, o caso mais raro teve um total de 94 ocorrências.

In [562]:
N = VFinal[VFinal['2020'] <2]
N

Junção  2019  2020
Acostamento em desnível                                  1   0.0     1
Ausência de sinalização                                  1   0.0     1
Afundamento ou ondulação no pavimento                    1   0.0     1
Frear bruscamente                                        1   0.0     1
Ingestão de álcool ou de substâncias psicoativa...       1   0.0     1
Acumulo de água sobre o pavimento                        1   0.0     1
Trafegar com motocicleta (ou similar) entre as ...       1   0.0     1
Condutor usando celular                                  1   0.0     1
Acumulo de óleo sobre o pavimento                        1   0.0     1
Pedestre cruzava a pista fora da faixa                   1   0.0     1

a3) Acima temos os casos mais raros de 2020 e da junção dos dois anos, obtidos por filtragem. Agora abaixo, temos o caso mais raro de 2019.

In [563]:
N = VFinal[VFinal['2019'] != 0]
N.tail(1)

Junção  2019  2020
Ingestão de Substâncias Psicoativas     180  94.0    86

b) O primeiro passo para rastrear e filtrar a contagem por meses, é transformar o Dtype da coluna de datas em Datetime. Em seguida os dados foram separados por Mês, para então ocorrer o cálculo da proporção de pessoas ilesas e de feridos graves de cada mês. Os dados foram colocados em uma lista à parte. O primeiro processamento foi dos dados de 2019, o segundo, similar ao primeiro, foram os de 2020, também colocados em uma lista à parte. No final, juntaram-se as duas listas em um dicionário para formar um novo Data frame.

In [564]:
V1 = pd.read_csv('https://raw.githubusercontent.com/Carol-ACDR/datatran2019-2020/main/datatran2019.csv',sep =';',encoding='latin_1')
V1['data_inversa'] = pd.to_datetime(V1['data_inversa'])
V1['data_inversa'] = V1['data_inversa'].dt.month
L = []
F = V1[['data_inversa','pessoas','feridos_graves','ilesos']]
Jan1 = F[F['data_inversa'] == 1]
Fev1 = F[F['data_inversa'] == 2]
Mar1 = F[F['data_inversa'] == 3]
Abr1 = F[F['data_inversa'] == 4]
Mai1 = F[F['data_inversa'] == 5]
Jun1 = F[F['data_inversa'] == 6]
Jul1 = F[F['data_inversa'] == 7]
Ags1 = F[F['data_inversa'] == 8]
Sep1 = F[F['data_inversa'] == 9]
Out1 = F[F['data_inversa'] == 10]
Nov1 = F[F['data_inversa'] == 11]
Des1 = F[F['data_inversa'] == 12]
P1 = L.append(((Jan1['ilesos'].sum()+Jan1['feridos_graves'].sum())/Jan1['pessoas'].sum())*100)
P2 =L.append(((Fev1['ilesos'].sum()+Fev1['feridos_graves'].sum())/Fev1['pessoas'].sum())*100)
P3 =L.append(((Mar1['ilesos'].sum()+Mar1['feridos_graves'].sum())/Mar1['pessoas'].sum())*100)
P4 =L.append(((Abr1['ilesos'].sum()+Abr1['feridos_graves'].sum())/Abr1['pessoas'].sum())*100)
P5 =L.append(((Mai1['ilesos'].sum()+Mai1['feridos_graves'].sum())/Mai1['pessoas'].sum())*100)
P6 =L.append(((Jun1['ilesos'].sum()+Jun1['feridos_graves'].sum())/Jun1['pessoas'].sum())*100)
P7 =L.append(((Jul1['ilesos'].sum()+Jul1['feridos_graves'].sum())/Jul1['pessoas'].sum())*100)
P8 =L.append(((Ags1['ilesos'].sum()+Ags1['feridos_graves'].sum())/Ags1['pessoas'].sum())*100)
P9 =L.append(((Sep1['ilesos'].sum()+Sep1['feridos_graves'].sum())/Sep1['pessoas'].sum())*100)
P10 =L.append(((Out1['ilesos'].sum()+Out1['feridos_graves'].sum())/Out1['pessoas'].sum())*100)
P11 =L.append(((Nov1['ilesos'].sum()+Nov1['feridos_graves'].sum())/Nov1['pessoas'].sum())*100)
P12 =L.append(((Des1['ilesos'].sum()+Des1['feridos_graves'].sum())/Des1['pessoas'].sum())*100)
L

[53.748760582716805,
 53.08924485125858,
 54.77927629154691,
 52.16543113538822,
 53.59124658069558,
 54.53732011035717,
 55.34854511199497,
 52.97667833991506,
 53.172050869755886,
 53.87265810063886,
 53.68588584243652,
 53.666987487969195]

In [565]:
V2 = pd.read_csv('https://raw.githubusercontent.com/Carol-ACDR/datatran2019-2020/main/datatran2020.csv',sep =';',encoding='latin_1')
V2['data_inversa'] = pd.to_datetime(V2['data_inversa'])
V2['data_inversa'] = V2['data_inversa'].dt.month
H = []
F = V2[['data_inversa','pessoas','feridos_graves','ilesos']]
Jan2 = F[F['data_inversa'] == 1]
Fev2 = F[F['data_inversa'] == 2]
Mar2 = F[F['data_inversa'] == 3]
Abr2 = F[F['data_inversa'] == 4]
Mai2 = F[F['data_inversa'] == 5]
Jun2 = F[F['data_inversa'] == 6]
Jul2 = F[F['data_inversa'] == 7]
Ags2 = F[F['data_inversa'] == 8]
Sep2 = F[F['data_inversa'] == 9]
Out2 = F[F['data_inversa'] == 10]
Nov2 = F[F['data_inversa'] == 11]
Des2 = F[F['data_inversa'] == 12]
P1 = H.append(((Jan2['ilesos'].sum()+Jan2['feridos_graves'].sum())/Jan2['pessoas'].sum())*100)
P2 =H.append(((Fev2['ilesos'].sum()+Fev2['feridos_graves'].sum())/Fev2['pessoas'].sum())*100)
P3 =H.append(((Mar2['ilesos'].sum()+Mar2['feridos_graves'].sum())/Mar2['pessoas'].sum())*100)
P4 =H.append(((Abr2['ilesos'].sum()+Abr2['feridos_graves'].sum())/Abr2['pessoas'].sum())*100)
P5 =H.append(((Mai2['ilesos'].sum()+Mai2['feridos_graves'].sum())/Mai2['pessoas'].sum())*100)
P6 =H.append(((Jun2['ilesos'].sum()+Jun2['feridos_graves'].sum())/Jun2['pessoas'].sum())*100)
P7 =H.append(((Jul2['ilesos'].sum()+Jul2['feridos_graves'].sum())/Jul2['pessoas'].sum())*100)
P8 =H.append(((Ags2['ilesos'].sum()+Ags2['feridos_graves'].sum())/Ags2['pessoas'].sum())*100)
P9 =H.append(((Sep2['ilesos'].sum()+Sep2['feridos_graves'].sum())/Sep2['pessoas'].sum())*100)
P10 =H.append(((Out2['ilesos'].sum()+Out2['feridos_graves'].sum())/Out2['pessoas'].sum())*100)
P11 =H.append(((Nov2['ilesos'].sum()+Nov2['feridos_graves'].sum())/Nov2['pessoas'].sum())*100)
P12 =H.append(((Des2['ilesos'].sum()+Des2['feridos_graves'].sum())/Des2['pessoas'].sum())*100)
H

[52.30277185501067,
 53.136615337935226,
 51.34608243594504,
 49.427207637231504,
 51.70838299562972,
 53.9381759020002,
 53.29547410084323,
 54.23820992929842,
 53.367246222762134,
 53.066298342541444,
 52.772608857461854,
 53.416605437178546]

In [567]:
Chaves = {'Porcentagem_ilesos+feridos-Gravemente_2019': L,'Porcentagem_ilesos+feridos-Gravemente_2020': H}
Proporção = pd.DataFrame(Chaves, index = ['Janeiro','Fevereiro','Março','Abril','Maio','Junho','Julho','Agosto','Setembro','Outubro','Novembro','Dezembro'])
Proporção

Porcentagem_ilesos+feridos-Gravemente_2019  \
Janeiro                                     53.748761   
Fevereiro                                   53.089245   
Março                                       54.779276   
Abril                                       52.165431   
Maio                                        53.591247   
Junho                                       54.537320   
Julho                                       55.348545   
Agosto                                      52.976678   
Setembro                                    53.172051   
Outubro                                     53.872658   
Novembro                                    53.685886   
Dezembro                                    53.666987   

           Porcentagem_ilesos+feridos-Gravemente_2020  
Janeiro                                     52.302772  
Fevereiro                                   53.136615  
Março                                       51.346082  
Abril                                       49.427208  
Maio                                        51.708383  
Junho                                       53.938176  
Julho                                       53.295474  
Agosto                                      54.238210  
Setembro                                    53.367246  
Outubro                                     53.066298  
Novembro                                    52.772609  
Dezembro                                    53.416605

c) Primeiramente, concatenou-se as séries de dados e seu total com o intuito de uma melhor visualização. No entanto, ainda sim se é difícil dizer, pois já de início os dados demonstram tanto aumentos quanto quedas em diferentes rodovias. Para solucionar tal problema, foi a criação de outra coluna, a qual demonstrava o número de casos que haviam aumentado ou diminuído.

In [568]:
V1 = d2019
V2 = d2020
RF19 = d2019['br']
RF20 = d2020['br']
RFT = pd.concat([RF19,RF20])
Junção = RFT.value_counts()
RFinal = pd.concat([Junção,RF19.value_counts(),RF20.value_counts()], axis = 1, keys =['Total','2019','2020'])
RFinal.fillna(0, inplace = True)
RFinal

Total    2019    2020
10.0     754   366.0   388.0
20.0    1897   933.0   964.0
30.0      15     7.0     8.0
40.0    6259  3244.0  3015.0
50.0    1449   687.0   762.0
...      ...     ...     ...
485.0      5     0.0     5.0
487.0     44    17.0    27.0
488.0     54    30.0    24.0
493.0    451   261.0   190.0
495.0     35    19.0    16.0

[119 rows x 3 columns]

In [569]:
RFinal['Aumentos e baixas'] =  RFinal['2020'] - RFinal['2019'] 
RFinal

Total    2019    2020  Aumentos e baixas
10.0     754   366.0   388.0               22.0
20.0    1897   933.0   964.0               31.0
30.0      15     7.0     8.0                1.0
40.0    6259  3244.0  3015.0             -229.0
50.0    1449   687.0   762.0               75.0
...      ...     ...     ...                ...
485.0      5     0.0     5.0                5.0
487.0     44    17.0    27.0               10.0
488.0     54    30.0    24.0               -6.0
493.0    451   261.0   190.0              -71.0
495.0     35    19.0    16.0               -3.0

[119 rows x 4 columns]

In [570]:
Maior19 = RFinal[RFinal['2019'] > RFinal['2020']]
Maior20 = RFinal[RFinal['2020'] > RFinal['2019']]
Resul = len(Maior19) - len(Maior20)

Resul_Total = RFinal['Aumentos e baixas'].sum()

print(Resul,Resul_Total)

24 -4059.0


c) Em suma, ocorreu uma diminuição de incidência de acidentes em 24 rodovias federais. Uma diminuição de 4059 entradas. Podemos supor que foi por conta da pandemia.

In [571]:
RFinal['Aumentos e baixas'] =  RFinal['2020'] - RFinal['2019'] 
RFinal


Total    2019    2020  Aumentos e baixas
10.0     754   366.0   388.0               22.0
20.0    1897   933.0   964.0               31.0
30.0      15     7.0     8.0                1.0
40.0    6259  3244.0  3015.0             -229.0
50.0    1449   687.0   762.0               75.0
...      ...     ...     ...                ...
485.0      5     0.0     5.0                5.0
487.0     44    17.0    27.0               10.0
488.0     54    30.0    24.0               -6.0
493.0    451   261.0   190.0              -71.0
495.0     35    19.0    16.0               -3.0

[119 rows x 4 columns]